src/core/Akka/Helios.Concurrency.DedicatedThreadPool.cs

```diff
@@ -237,27 +245,32 @@ private void RequestWorker()
                     while (true)
                     {
                         Task item;
-                        lock (_tasks)
+                        lock (_scheduler._tasks)
                         {
                             // done processing
-                            if (_tasks.Count == 0)
+                            if (_scheduler._tasks.Count == 0)
                             {
-                                ReleaseWorker();
+                                _scheduler.ReleaseWorker();
                                 break;
                             }
 
                             // Get the next item from the queue
-                            item = _tasks.First.Value;
-                            _tasks.RemoveFirst();
+                            item = _scheduler._tasks.First.Value;
+                            _scheduler._tasks.RemoveFirst();
                         }
 
                         // Execute the task we pulled out of the queue
-                        TryExecuteTask(item);
+                        _scheduler.TryExecuteTask(item);
                     }
                 }
                 // We're done processing items on the current thread
                 finally { _currentThreadIsRunningTasks = false; }
-            });
+            }
+        }
+
+        private void RequestWorker()
+        {
+            _pool.QueueUserWorkItem(new RequestWorkerTask(this));
```

Comment:
We end up boxing this struct and paying for the allocation. We could get fancy w/ generic type constraints to get around it.

``` csharp
public bool QueueUserWorkItem<T>(T work) where T : IRunnable
{
    // same code
}
```

Reply:
That's a good idea - I'd be happy to do that or you can send in a PR.